src/protocol/server.c

```diff
@@ -92,17 +90,50 @@ protocolServerError(ProtocolServer *this, int code, const String *message, const
     ASSERT(message != NULL);
     ASSERT(stack != NULL);
 
-    KeyValue *error = kvNew();
-    kvPut(error, VARSTR(PROTOCOL_ERROR_STR), VARINT(code));
-    kvPut(error, VARSTR(PROTOCOL_OUTPUT_STR), VARSTR(message));
-    kvPut(error, VARSTR(PROTOCOL_ERROR_STACK_STR), VARSTR(stack));
+    // Write the error and flush to be sure it gets sent immediately
+    PackWrite *error = pckWriteNew(this->write);
+    pckWriteU32P(error, protocolMessageTypeError);
+    pckWriteI32P(error, code);
+    pckWriteStrP(error, message);
+    pckWriteStrP(error, stack);
+    pckWriteEndP(error);
 
-    ioWriteStrLine(protocolServerIoWrite(this), jsonFromKv(error));
-    ioWriteFlush(protocolServerIoWrite(this));
+    ioWriteFlush(this->write);
```

Comment:
The Pack is initialized as `pckWriteNew(this->write)`. This tells it to use the IoWrite interface to write directly to the client (instead of to a local buffer). At the end we flush the same IoWrite interface to be sure there is no data sitting in a buffer somewhere.